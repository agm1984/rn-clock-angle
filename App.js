import React, { Component } from 'react'
import { View, Text, TextInput, StyleSheet } from 'react-native'
import Clock from './src/components/Clock/Clock'

const styles = StyleSheet.create({
  appWrapper: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'stretch',
    justifyContent: 'center',
  },
  appContainer: {
    alignItems: 'stretch',
  },
  appHeading: {
    backgroundColor: '#000',
    color: '#fff',
    textAlign: 'center',
    fontSize: 32,
  },
  currentTimeWrapper: {
    flex: 0,
    flexDirection: 'row',
    justifyContent: 'center',
    backgroundColor: 'brown',
  },
  hoursInput: {
    height: 40,
    borderColor: 'gray',
    borderWidth: 1,
    textAlign: 'center',
    padding: 10,
    margin: 10,
  },
  minuteInput: {
    height: 40,
    borderColor: 'gray',
    borderWidth: 1,
    textAlign: 'center',
    padding: 10,
    margin: 10,
  },
})

class App extends Component {
  constructor(props) {
    super(props)
    const d = new Date()
    // this.state = {
    //   sec: d.getSeconds() * 6,
    //   min: d.getMinutes() * 6 + (d.getSeconds() * 6) / 60,
    //   hour: ((d.getHours() % 12)/ 12) * 360 + 90 +
    //     (d.getMinutes() * 6 + (d.getSeconds() * 6) / 60) / 12,
    // }
    this.state = {
      currentHour: d.getHours().toString(),
      currentMinute: d.getMinutes().toString(),
      primaryColor: 'black',
    }
  }

  render() {
    const { currentHour, currentMinute, primaryColor } = this.state
    return (
      <View style={styles.appWrapper}>
        <View style={styles.appContainer}>
          <Text style={styles.appHeading}>
            [%] INANGLES
          </Text>
          <View style={styles.currentTimeWrapper}>
            <TextInput
              style={styles.hoursInput}
              placeholder={currentHour}
              onChangeText={text => this.setState({ currentHour: text })}
              value=""
            />
            <TextInput
              style={styles.minuteInput}
              placeholder={currentMinute}
              onChangeText={text => this.setState({ currentMinute: text })}
              value=""
            />
          </View>
          <Clock
            clockSize={270}
            clockBorderWidth={7}
            clockCentreSize={15}
            clockCentreColor={primaryColor}
            hourHandColor={primaryColor}
            hourHandWidth={5.5}
            hourHandLength={70}
            hourHandCurved
            hourHandOffset={0}
            minuteHandColor={primaryColor}
            minuteHandLength={110}
            minuteHandWidth={5}
            minuteHandCurved
            minuteHandOffset={0}
            secondHandColor={primaryColor}
            secondHandLength={120}
            secondHandWidth={2}
            secondHandCurved
            secondHandOffset={0}
          />
        </View>
      </View>
    )
  }
}

export default App
