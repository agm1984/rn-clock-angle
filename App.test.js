import React from 'react'
import renderer from 'react-native-test-utils'
import App from './App'

describe('./App.js', () => {
  const app = renderer(<App />)
  const title = '[%] INANGLES'
  const d = new Date()
  const currentHour = d.getHours().toString()
  const currentMinute = d.getMinutes().toString()
  // if (currentHour > 12)

  it('renders without crashing', () => {
    expect(app).toBeTruthy()
  })
  it('has the correct title', () => {
    const appHeading = app.query('Text')
    expect(appHeading.text()).toEqual(title)
  })

  describe('Hour Input', () => {
    it('has the correct initial state', () => {
      expect(app.state().currentHour).toEqual(currentHour)
    })
    it('displays current hour as placeholder', () => {
      const textInputView = app.query(`TextInput[placeholder='${currentHour}']`)
      textInputView.simulate('changeText', 'react-native-test-utils')
    })
  })

  describe('Minute Input', () => {
    it('has the correct initial state', () => {
      expect(app.state().currentMinute).toEqual(currentMinute)
    })
    it('displays current minute as placeholder', () => {
      const textInputView = app.query(`TextInput[placeholder='${currentMinute}']`)
      textInputView.simulate('changeText', 'react-native-test-utils')
    })
  })
})
