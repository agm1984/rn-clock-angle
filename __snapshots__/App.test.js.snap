// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./App.js renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "#000",
          "color": "#fff",
          "fontSize": 32,
          "textAlign": "center",
        }
      }
    >
      [%] INANGLES
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "brown",
          "flex": 0,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="15"
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "margin": 10,
            "padding": 10,
            "textAlign": "center",
          }
        }
        value=""
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="49"
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "margin": 10,
            "padding": 10,
            "textAlign": "center",
          }
        }
        value=""
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "orange",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#eee",
            "borderColor": "black",
            "borderRadius": 135,
            "borderWidth": 7,
            "height": 270,
            "position": "relative",
            "width": 270,
          }
        }
      >
        <View
          style={
            Object {
              "bottom": -7,
              "height": 270,
              "position": "absolute",
              "right": -7,
              "width": 270,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "black",
                  "borderBottomLeftRadius": 5.5,
                  "borderTopLeftRadius": 5.5,
                  "height": 0,
                  "left": 135,
                  "marginLeft": -35,
                  "marginVertical": -5.5,
                  "paddingLeft": 70,
                  "paddingVertical": 5.5,
                  "position": "absolute",
                  "top": 135,
                  "width": 0,
                },
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "204.64166666666665deg",
                    },
                    Object {
                      "translateX": -35,
                    },
                  ],
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "black",
                  "borderTopLeftRadius": 5,
                  "borderTopRightRadius": 5,
                  "height": 0,
                  "left": 135,
                  "marginHorizontal": -5,
                  "marginTop": -55,
                  "paddingHorizontal": 5,
                  "paddingTop": 110,
                  "position": "absolute",
                  "top": 135,
                  "width": 0,
                },
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "295.7deg",
                    },
                    Object {
                      "translateY": -55,
                    },
                  ],
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "black",
                  "borderTopLeftRadius": 2,
                  "borderTopRightRadius": 2,
                  "height": 0,
                  "left": 135,
                  "marginHorizontal": -2,
                  "marginTop": -60,
                  "paddingHorizontal": 2,
                  "paddingTop": 120,
                  "position": "absolute",
                  "top": 135,
                  "width": 0,
                },
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "102deg",
                    },
                    Object {
                      "translateY": -60,
                    },
                  ],
                },
              ]
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "black",
                "borderRadius": 7.5,
                "height": 15,
                "left": 127.5,
                "top": 127.5,
                "width": 15,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
